provider.go

```diff
@@ -854,13 +820,138 @@ func (p *provider) Call(ctx context.Context, req *rpc.CallRequest) (*rpc.CallRes
 	}, nil
 }
 
+func (h *host) Call(ctx context.Context, req CallRequest, call comProvider.CallFunc,
+) (CallResponse, error) {
+	r, err := comProvider.Call(ctx, req.rpc(internalrpc.MarshalProperties), h.host.EngineConn(), call)
+	if err != nil {
+		return CallResponse{}, err
+	}
+	return newCallResponse(r)
+}
+
 // CallRequest represents a requested resource method invocation.
 //
 // It corresponds to [rpc.CallRequest] on the wire.
 type CallRequest struct {
-	Tok     tokens.ModuleMember
-	Args    presource.PropertyMap
-	Context *pulumi.Context
+	// the function token to invoke.
+	Tok tokens.ModuleMember
+	// the arguments for the function invocation.
```

Comment:
Possible nit: `s/function/method/g`?

Reply:
I got the text from upstream:
https://github.com/pulumi/pulumi/blob/0ff190607702431a286161d8847902e2238e39f2/proto/pulumi/provider.proto#L502-L512